expression/aggregation/base_func.go

```diff
@@ -443,5 +444,31 @@ func (a *baseFuncDesc) WrapCastForAggArgs(ctx sessionctx.Context) {
 		originTp := a.Args[i].GetType().Tp
 		*(a.Args[i].GetType()) = *(a.RetTp)
 		a.Args[i].GetType().Tp = originTp
+		// refine each mysql integer type to the needed decimal precision for sum
+		if a.Name == ast.AggFuncSum && types.IsTypeInteger(tpOld) {
+			if flen, err := refineSumDecimalArgs(tpOld); err != nil {
+				// avg could be split into sum and count, so we should take float decimal into account
+				a.Args[i].GetType().Flen = mathutil.Min(a.Args[i].GetType().Flen, flen+a.Args[i].GetType().Decimal)
```

Comment:
also explain the `mathutil.Min`?

Reply:
just integer min.
it's already in this file before this PR.